exchanges/persisted-fetch/src/sha256.ts

```diff
@@ -30,61 +30,56 @@ const sha256Browser = (bytes: Uint8Array): Promise<Uint8Array> => {
   });
 };
 
-let nodeCrypto;
-if (typeof window === 'undefined') {
+let nodeCrypto: Promise<typeof import('crypto')> | void;
+if (typeof window === 'undefined' && !webCrypto) {
+  // Indirect eval'd require/import to guarantee no side-effects in module scope
+  // (optimization for minifiers)
   try {
-    // Indirect eval/require to guarantee no side-effects in module scope
-    // (optimization for minifiers)
-    nodeCrypto = new Function('require', 'return require("crypto")')(require);
-  } catch (e) {}
+    nodeCrypto = Promise.resolve(new Function('return require("crypto")')());
+  } catch (_error) {
+    // @ts-expect-error
+    try {
+      nodeCrypto = import('crypto');
```

Comment:
Are bundlers still going to try and add `crypto` to the bundle with this present (even deeply inside try/catches)?

(I should mention that haven't actually experimented with the build end of things and inspected any bundles. 😂 )

Reply:
Actually, let's make that one indirect too, just in case. I didn't test this thoroughly, so who knows what happens in other bundlers 🤔 I can totally see this causing problems when it isn't treated as a Node builtin.